### PR TITLE
Docs: Improve examples for no-case-declarations (fixes #6716)

### LINKED ISSUE #6716: no-case-declarations examples are very similar and hard to distinguish
Hey, 

looking at http://eslint.org/docs/rules/no-case-declarations#rule-details I cannot really see the difference between correct and incorrect code. Either I am missing something or the declarations should be removed there.

### PR DESCRIPTION
**Pull request by rangelke**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

PR for the following issue https://github.com/eslint/eslint/issues/6716

**What changes did you make? (Give an overview)**
I added two examples where no-case-declarations are used correctly, one declaration outside the switch-statement and one using var. 
The first the block scope of a case is introduced I added a comment to make it clearer where brackets should be used.

**Is there anything you'd like reviewers to focus on?**
I have troubles with an appropriate commit message. Is it too vague? If so, any proposals, since it's hard to keep in the 72 character limit?

### CODE REVIEW

File: docs/rules/no-case-declarations.md [hunk 1]
```diff
@@ -39,7 +39,11 @@ Examples of **correct** code for this rule:
 /*eslint no-case-declarations: "error"*/
 /*eslint-env es6*/
 
+/* Declarations outside switch-statements are valid */
```
Comment (arv) [inline docs/rules/no-case-declarations.md]: By looking at other example code I think it is more common to use `//` comments.

Comment (rangelke) [inline docs/rules/no-case-declarations.md]: I'll update!

**Comment by rangelke** [comment]
@arv The appveyor build failed, could it be because of the name change of my commit?

**Comment by vitorbal** [comment]
@rangelke Thanks, this looks good! The appveyor build seems green, maybe it just wasn't finished yet when you checked.

**Pull request merged**